src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -216,7 +216,7 @@ public String getRemoteFs() {
     }
 
     public void setInstanceCap(int instanceCap) {
-        if (instanceCap <= 0) {
+        if (instanceCap < 0) {
```

Comment:
Maybe I am wrong, but doesn't this also need to go into [`readResolve`](https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility) for backward compatibility?